### PR TITLE
[tests] Add .NET version of link sdk for macOS.

### PR DESCRIPTION
**Pull request by rolfbjarne**
This required adding a linker substitution file for macOS (and I also added one for Mac Catalyst).

### CODE REVIEW

File: tests/linker/ios/link sdk/CryptoTest.cs [hunk 1]
```diff
@@ -182,7 +182,12 @@ public void Chain ()
 			Assert.False (chain.Build (cert), "Offline");
 
 			chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+#if __MACOS__
+			// Not sure if this is expected on macOS, or if it's a bug somewhere in our code.
+			Assert.False (chain.Build (cert), "NoCheck");
```
Comment (spouliot) [inline tests/linker/ios/link sdk/CryptoTest.cs:187]: does that happen on both mono and dotnet runtimes ?

**Review by spouliot — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tests/linker/ios/link sdk/Bug2096Test.cs [hunk 2]
```diff
@@ -44,10 +44,12 @@ public void Bug2096_Aot ()
 			
 			var x = new D ();
 			a.OuterMethod<D> (x);
+#if !__MACOS__
 			// fails with 5.0.2 (5.1) when running on devices with
 			// Attempting to JIT compile method 'A:InnerMethod<D, long> (D,long)' while running with --aot-only.
 			if (Runtime.Arch == Arch.SIMULATOR)
 				Assert.Inconclusive ("only fails on devices");
+#endif
```
Comment (mandel-macaque) [inline tests/linker/ios/link sdk/Bug2096Test.cs:52]: I wonder if we could move this to a common name and do a noop  on mac and do assert in iOS and others.

**Review by mandel-macaque — APPROVED** [review]

File: tests/linker/ios/link sdk/CryptoTest.cs (hunk 1 above)
Comment (rolfbjarne) [reply tests/linker/ios/link sdk/CryptoTest.cs:187]: Only .NET.

I'll create a test case and file an issue in dotnet/runtime.

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* monotouch-test/Mac Catalyst/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2304
* link all/Mac Catalyst/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2414

File: tests/linker/ios/link sdk/CryptoTest.cs (hunk 1 above)
Comment (rolfbjarne) [reply tests/linker/ios/link sdk/CryptoTest.cs:187]: That said, the other .NET platforms hit this case:

https://github.com/xamarin/xamarin-macios/blob/213dbd2c1d7e835a062313dd94d01f28e24d0052/tests/linker/ios/link%20sdk/CryptoTest.cs#L162

Comment (rolfbjarne) [reply tests/linker/ios/link sdk/CryptoTest.cs:187]: https://github.com/dotnet/runtime/issues/57630

**Pull request merged**